bundles/org.openhab.binding.dbquery/src/test/java/org/openhab/binding/dbquery/internal/Value2StateConverterTest.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Test
+    void given_date_and_datetime_target_return_datetype() {
+        var date = new Date();
+        var converted = instance.convertValue(date, DateTimeType.class);
+        assertThat(converted, instanceOf(DateTimeType.class));
+        assertThat(((DateTimeType) converted).getZonedDateTime(),
+                is(ZonedDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault()).withFixedOffsetZone()));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "2019-10-12T07:20:50.52Z", "2019-10-12" })
+    void given_valid_string_date_and_datetime_target_return_datetype(String date) {
+        var converted = instance.convertValue(date, DateTimeType.class);
+        assertThat(converted, instanceOf(DateTimeType.class));
+        var convertedDateTime = ((DateTimeType) converted).getZonedDateTime();
+        assertThat(convertedDateTime.getYear(), is(2019));
+        assertThat(convertedDateTime.getMonthValue(), is(10));
+        assertThat(convertedDateTime.getDayOfMonth(), is(12));
+        assertThat(convertedDateTime.getHour(), anyOf(is(7), is(0)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("trueValues")
+    void given_values_considerated_true_and_on_off_target_return_on(Object value) {
+        var converted = instance.convertValue(value, OnOffType.class);
+        assertThat(converted, instanceOf(OnOffType.class));
+        assertThat(converted, is(OnOffType.ON));
+    }
+
+    @ParameterizedTest
+    @MethodSource("falseValues")
+    void given_values_considerated_false_and_on_off_target_return_off(Object value) {
+        var converted = instance.convertValue(value, OnOffType.class);
+        assertThat(converted, instanceOf(OnOffType.class));
+        assertThat(converted, is(OnOffType.OFF));
+    }
+
+    @ParameterizedTest
+    @MethodSource("trueValues")
+    void given_values_considerated_true_and_open_closed_target_return_open(Object value) {
+        var converted = instance.convertValue(value, OpenClosedType.class);
+        assertThat(converted, instanceOf(OpenClosedType.class));
+        assertThat(converted, is(OpenClosedType.OPEN));
+    }
+
+    @ParameterizedTest
+    @MethodSource("falseValues")
+    void given_values_considerated_false_and_open_closed_target_return_closed(Object value) {
+        var converted = instance.convertValue(value, OpenClosedType.class);
+        assertThat(converted, instanceOf(OpenClosedType.class));
+        assertThat(converted, is(OpenClosedType.CLOSED));
+    }
+
+    private static Stream<Object> trueValues() {
+        return Stream.of("true", "True", 1, 2, "On", "on", -1, 0.3);
+    }
+
+    private static Stream<Object> falseValues() {
+        return Stream.of("false", "False", 0, 0.0d, "off", "Off", "", "a value");
+    }
+
+    private static Stream<Number> provideNumericTypes() {
+        return Stream.of(1L, 1.2, 1.2f, -1, 0, new BigDecimal("212321213123123123123123"));
```

Comment:
How often does this run? Consider creating a static BigDecimal and reusing it. You are also using a similar line of code a new lines lower down in this same file. This appears to be parsing a String to create the BigDecimal when you can normally provide a number instead of a string to achieve the same thing.

Reply:
Done the static part, about the string it's used because I want to test a real big number that can't be expressed in a long literal string.
As this is a test I think that is best to test edge cases than premature optimization